c7n/resources/servicecatalog.py

```diff
@@ -0,0 +1,154 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+
+from c7n.actions import BaseAction
+from c7n.filters import CrossAccountAccessFilter
+from c7n.query import ConfigSource, DescribeSource, QueryResourceManager, TypeInfo
+from c7n.manager import resources
+from c7n.exceptions import PolicyValidationError
+from c7n.tags import universal_augment
+from c7n.utils import local_session, type_schema
+
+
+class DescribePortfolio(DescribeSource):
+
+    def augment(self, resources):
+        return universal_augment(self.manager, super().augment(resources))
+
+
+@resources.register('catalog-portfolio')
+class CatalogPortfolio(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'servicecatalog'
+        enum_spec = ('list_portfolios', 'PortfolioDetails', None)
+        detail_spec = ('describe_portfolio', 'Id', 'Id', None)
+        arn = 'ARN'
+        id = 'Id'
+        name = 'DisplayName'
+        date = 'CreatedTime'
+        universal_taggable = object()
+        cfn_type = config_type = 'AWS::ServiceCatalog::Portfolio'
+
+    source_mapping = {
+        'describe': DescribePortfolio,
+        'config': ConfigSource
+    }
+
+
+@CatalogPortfolio.action_registry.register('delete')
+class CatalogPortfolioDeleteAction(BaseAction):
+    """Action to delete a Service Catalog Portfolio
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: service-catalog-portfolio-delete
+            resource: aws.catalog-portfolio
+            filters:
+              - type: cross-account
+            actions:
+              - type: remove-shared-accounts
```

Comment:
Should we drop `remove-shared-accounts` from this example as it is focused on the `delete` action?